src/multiselect_contextmenu.js

```diff
@@ -702,30 +755,30 @@ const registerSelectAll = function() {
              !block.isInsertionMarker();
     },
     callback: function(scope) {
-      const blockSelection = blockSelectionWeakMap.get(scope.workspace);
-      if (Blockly.getSelected() &&
-        !blockSelection.has(Blockly.getSelected().id)) {
+      const multiDraggable = multiDraggableWeakMap.get(scope.workspace);
+
+      // Make sure that there is nothing in the multiDraggable (clearing)
+      // prior to selecting all blocks in workspace.
+      if (Blockly.getSelected()) {
         Blockly.getSelected().pathObject.updateSelected(false);
         Blockly.common.setSelected(null);
+        multiDraggable.clearAll_();
       }
       const blockList = [];
       scope.workspace.getTopBlocks().forEach(function(block) {
         if (selectAllOption.check(block)) {
           blockList.push(block);
-          let nextBlock = block.getNextBlock();
-          while (nextBlock) {
-            blockList.push(nextBlock);
-            nextBlock = nextBlock.getNextBlock();
-          }
```

Comment:
Why did you get this removed? This is clearly causing another user behavior inconsistency. We should get all the connected statements selected, or we won't be able to duplicate/copy all the blocks in workspace:

![Jul-03-2024 10-18-28](https://github.com/mit-cml/workspace-multiselect/assets/43995067/ee074a1e-8e74-4ee0-b4f9-e84fcea652fb)

Reply:
I am not sure why this was removed. Maybe I was thinking it was related to some bug. I added it back and will continue to test it out. So far, it looks like it is working as intended.